sig-provider/src/aggregator.rs

```diff
@@ -0,0 +1,96 @@
+use crate::{
+    proto::blockscout::sig_provider::v1::{
+        signature_service_server::SignatureService, CreateSignaturesRequest,
+        CreateSignaturesResponse, GetSignaturesRequest, GetSignaturesResponse, Signature,
+    },
+    provider::SignatureProvider,
+};
+use std::sync::Arc;
+
+pub struct SignatureAggregator {
+    sources: Vec<Arc<dyn SignatureProvider + Send + Sync + 'static>>,
+}
+
+impl SignatureAggregator {
+    pub fn new(
+        sources: Vec<Arc<dyn SignatureProvider + Send + Sync + 'static>>,
```

Comment:
Maybe rename all `provider` to `source` ? I mean, name of this microservice is `sig-provider`, so in order not to confuse this, it is better to call the 4bytes and sigeth structures as the "SignatureSource"